spring-test/src/main/java/org/springframework/test/web/client/match/MultipartFormDataRequestMatchers.java

```diff
@@ -0,0 +1,252 @@
+					Assert.state(actualValues != null, "No values for request parameter " + param);
+
+					assertValueCount(param, requestParams, values.size());
+
+					assertEquals("Parameter " + param, values, Arrays.asList(actualValues));
+				});
+			}
+		};
+	}
+
+	public RequestMatcher file(String parameter, byte[]... resources) {
+		return new AbstractFormDataRequestMatcher() {
+			@Override
+			protected void matchInternal(MultipartHttpServletRequest request) {
+				MultiValueMap<String, MultipartFile> files = request.getMultiFileMap();
+				assertValueCount(parameter, files, resources.length);
+
+				assertByteArrayMatch(parameter, Arrays.asList(resources), files.get(parameter));
+			}
+		};
+	}
+
+	@SafeVarargs
+	@SuppressWarnings("varargs")
+	public final RequestMatcher file(String parameter, Matcher<? super Resource>... matchers) {
+		return new AbstractFormDataRequestMatcher() {
+			@Override
+			protected void matchInternal(MultipartHttpServletRequest request) {
+				MultiValueMap<String, MultipartFile> files = request.getMultiFileMap();
+				assertValueCount(parameter, files, matchers.length);
+				List<MultipartFile> parts = files.get(parameter);
+
+				for (int i = 0; i < matchers.length; i++) {
+					assertThat("File [" + parameter + "]", parts.get(i).getResource(), matchers[i]);
+				}
+			}
+		};
+	}
+
+	public RequestMatcher file(String parameter, Resource... resources) {
+		return new AbstractFormDataRequestMatcher() {
+			@Override
+			protected void matchInternal(MultipartHttpServletRequest request) {
+				MultiValueMap<String, MultipartFile> files = request.getMultiFileMap();
+				assertValueCount(parameter, files, resources.length);
+
+				assertResourceMatch(parameter, Arrays.asList(resources), files.get(parameter));
+			}
+		};
+	}
+
+	public RequestMatcher files(MultiValueMap<String, Resource> expectedFiles) {
+		return new AbstractFormDataRequestMatcher() {
+			@Override
+			protected void matchInternal(MultipartHttpServletRequest request) {
+				MultiValueMap<String, MultipartFile> actualFiles = request.getMultiFileMap();
+
+				expectedFiles.forEach((param, parts) -> {
+					assertValueCount(param, actualFiles, parts.size());
+					assertResourceMatch(param, parts, actualFiles.get(param));
+				});
+			}
+		};
+	}
+
+
+	private void assertByteArrayMatch(String parameterName, List<byte[]> expectedFiles, List<MultipartFile> actualFiles) {
+		for (int index = 0; index < actualFiles.size(); index++) {
+			MultipartFile multiPartFile = actualFiles.get(index);
+			byte[] expectedContent = expectedFiles.get(index);
+
+			try {
+				assertEquals("Content mismatch for file " + parameterName, expectedContent, multiPartFile.getBytes());
+			}
+			catch (IOException ex) {
+				throw new AssertionError("Could not get bytes from actual multipart files", ex);
+			}
+		}
+	}
+
+	private void assertResourceMatch(String parameterName, List<Resource> expectedFiles, List<MultipartFile> actualFiles) {
```

Comment:
This does not adhere to line length rules https://github.com/spring-projects/spring-framework/wiki/Code-Style#line-wrapping.